src/admin/resolvers/queries/ApprovedItem.ts

```diff
@@ -76,3 +77,53 @@ export async function getApprovedItemByUrl(
 
   return await dbGetApprovedItemByUrl(context.db, args.url);
 }
+
+/**
+ * Retrieves a history of when and on what scheduled surface a corpus item
+ * was scheduled.
+ */
+export async function getScheduledSurfaceHistory(
+  parent,
+  args,
+  { db }
+): Promise<ScheduledItem[]> {
+  // Get the external ID of the Approved Item itself
+  const { externalId } = parent;
+
+  // Get the optional filters. Not specifying any means we will retrieve
+  // the entire list of all the occasions this approved item was scheduled
+  // on all surfaces, in descending order (most recently scheduled first).
+  //
+  // Limiting it to a single surface means we'll only retrieve data for
+  // one scheduled surface (useful for when you only need the one to filter out
+  // prospects on the Prospecting page).
+  //
+  // Limiting the number of results to one means only the most recent result
+  // will be returned.
+  const { scheduledSurfaceGuid, limit } = args;
+
+  // Check if the scheduled surface supplied is valid
+  const surface = ScheduledSurfaces.find((surface) => {
+    return surface.guid === scheduledSurfaceGuid;
+  });
+  if (!surface) {
+    throw new UserInputError(
+      `Could not find Scheduled Surface with id of "${scheduledSurfaceGuid}".`
+    );
+  }
+
+  // Check if the limit specified passes basic sanity checks
+  if (limit.isNaN() || limit < 1) {
```

Comment:
wish we could have finer-grained argument validation on the graphql side easily... almost all numeric args should be > 0

Reply:
Thank you @kschelonka, that jogged my memory as I thought there was something out there to help with this. There's `NonNegativeInt` in `graphql-scalars`, trying it out on this field: https://www.graphql-scalars.dev/docs/scalars/non-negative-int